pkg/aws/v1alpha3/cluster/validate_cluster.go

```diff
@@ -119,6 +143,75 @@ func (v *Validator) ValidateUpdate(request *admissionv1.AdmissionRequest) (bool,
 	return true, nil
 }
 
+func (v *Validator) ClusterAnnotationUpgradeTimeIsValid(cluster *capiv1alpha3.Cluster) error {
+	if updateTime, ok := cluster.GetAnnotations()[annotation.UpdateScheduleTargetTime]; ok {
+		if !UpgradeScheduleTimeIsValid(updateTime) {
+			return microerror.Maskf(notAllowedError,
+				fmt.Sprintf("Cluster annotation '%s' value '%s' is not valid. Value must be in RFC822 format and should be a date in the future not more than 6 months away.",
+					annotation.UpdateScheduleTargetTime,
+					updateTime),
+			)
+		}
+	}
+	return nil
+}
+
+func UpgradeScheduleTimeIsValid(updateTime string) bool {
+	// parse time
+	t, err := time.Parse(time.RFC822, updateTime)
+	if err != nil {
+		return false
+	}
+	//time already passed
+	if t.Before(time.Now()) {
+		return false
+	}
+	// time too far in the future
+	if time.Until(t) > 4380*time.Hour {
```

Comment:
what is the reason of 4380 hours? maybe add this to the comment

Reply:
it's 6 months. And yeah, will add a comment.